src/openssl.c

```diff
@@ -1056,10 +1057,11 @@ _libssh2_cipher_crypt(_libssh2_cipher_ctx * ctx,
                decrypt: verify tag, if applicable
                in!=NULL is equivalent to EVP_CipherUpdate
                in==NULL is equivalent to EVP_CipherFinal */
+
 #ifdef HAVE_OPAQUE_STRUCTS
-            ret = EVP_Cipher(*ctx, NULL, NULL, 0); /* final */
```

Comment:
It should be fine to always call EVP_CipherFinal instead, no matter the backend.